requirements.txt

```diff
@@ -1,18 +1,38 @@
+attrs==19.3.0
+biopython==1.77
+certifi==2020.6.20
+chardet==3.0.4
+coverage==5.2.1
+coveralls==2.1.1
+Cython==0.29.21
 DateTime==4.3
-decorator==4.4.1
+decorator==4.4.2
 docopt==0.6.2
-jsonschema==3.0.0a3
-progressbar2==3.47.0
-py==1.8.1
-pytest==5.3.5
-pytest-cov==2.8.1
+idna==2.10
+iniconfig==1.0.1
+jsonschema==3.2.0
+more-itertools==8.4.0
+numpy==1.19.1
+packaging==20.4
+pandas==1.0.5
+parse==1.16.0
+pluggy==0.13.1
+progressbar2==3.51.4
+py==1.9.0
+pyparsing==2.4.7
+pyrsistent==0.16.0
+pytest==6.0.1
+pytest-cov==2.10.0
 pytest-runner==5.2
-pytz==2019.3
-requests==2.23.0
-requests_mock==1.7.0
+python-dateutil==2.8.1
+python-utils==2.4.0
+pytz==2020.1
+requests==2.24.0
+requests-mock==1.8.0
 retry==0.9.2
 setuptools-git==1.2
-sh==1.12.14
-six==1.14.0
-wheel==0.34.2
-zope.interface==4.7.1
\ No newline at end of file
+sh==1.13.1
+six==1.15.0
+toml==0.10.1
+urllib3==1.25.10
+zope.interface==5.1.0
```

Comment:
Only a minor. comment but this looks like the output of `pip freeze` or `pip list` which includes all the dependencies of your dependencies. I tend to prefer only including the first layer of dependencies because it's much easier to remove them afterward.

Reply:
I did consider that, but a very important point is that if you only specify first layer dependencies, then those can depend on the "latest" versions of other packages, so in this way you cannot really guarantee a reproducible behaviour. E.g. we only use Pandas explicitly and not Numpy. If we only pin Pandas version, then we are not protected against Numpy having a bad release and breaking things.

On the other hand, there are packages which visualise the PIP dependency tree, so when it's time to do cleanup, we should be able to use those.

I'm going to merge this PR as is due to time pressure, but please let me know if you feel strongly about dependencies, and then we could create a small follow-up issue and I would address it